components/sharing-dialog/src/sharing-dialog.js

```diff
@@ -29,269 +31,126 @@ const mutation = {
-            case 'public':
-                updatedAccess.public = access
-                break
-            case 'group':
-            case 'user': {
-                const pluralType = `${type}s`
-
-                updatedAccess[pluralType] = {
-                    ...sharingSettings[pluralType],
-                    [id]: {
-                        ...sharingSettings[pluralType][id],
-                        access,
-                    },
-                }
-                break
-            }
-        }
-
-        updateSharingSettings((prevState) => ({
-            ...prevState,
-            ...updatedAccess,
-        }))
-    }
-
-    const saveSharingSettings = (sharingSettings) => {
-        // prepare payload
-        const payload = {
-            object: {
-                publicAccess: convertConstantToAccess(sharingSettings.public),
-                externalAccess: convertConstantToAccess(
-                    sharingSettings.external,
-                    true
-                ),
-                userAccesses: Object.values(sharingSettings.users).map(
-                    (userSharingSettings) => ({
-                        ...userSharingSettings,
-                        access: convertConstantToAccess(
-                            userSharingSettings.access
-                        ),
-                    })
-                ),
-                userGroupAccesses: Object.values(sharingSettings.groups).map(
-                    (groupSharingSettings) => ({
-                        ...groupSharingSettings,
-                        access: convertConstantToAccess(
-                            groupSharingSettings.access
-                        ),
-                    })
-                ),
-            },
-        }
-
-        mutateSharingSettings(payload)
-    }
-
-    const onAdd = (data) => {
-        setIsDirty(true)
-
-        addUserOrGroupAccess(data)
-    }
-
-    const onChange = (data) => {
-        setIsDirty(true)
+        refetch({ type, id })
+    }, [type, id])
 
-        changeAccess(data)
+    /**
+     * Block interaction during the initial load
+     */
+
+    if (loading) {
+        const users = Object.keys(initialSharingSettings.users).map(mapUsers)
+        const groups = Object.keys(initialSharingSettings.groups).map(mapGroups)
+
+        return (
+            <FetchingContext.Provider value={true}>
+                <Modal onClose={onClose}>
+                    <TabbedContent
+                        id={id}
+                        users={users}
+                        groups={groups}
+                        publicAccess={initialSharingSettings.public}
+                        allowPublicAccess={initialSharingSettings.allowPublic}
+                        type={type}
+                        onAdd={() => {}}
+                        onChange={() => {}}
+                        onRemove={() => {}}
+                    />
+                </Modal>
+            </FetchingContext.Provider>
+        )
```

Comment:
I think that would be a good change yeah. One thing we should probably check though: what was the intention of the initialsharingsettings prop. It seems to me to just be a way to seed the component with data before it even does any fetching. If that's true, then ignoring initial sharing settings and showing a spinner instead _should_ be harmless.

But, technically we _are_ breaking the component by removing this functionality. Say someone has a test that asserts that passing in that prop will result in the component displaying the data, we would break that test. I think it's probably best to defer this change (which I'm still all for) and release it as a breaking change later. Maybe we could create a proposal that deprecates this in a separate PR?

Reply:
Agreed

Reply:
Just unresolving this so I don't forget to follow up.

Reply:
Added it to the top level follow ups.